scripts/py_matter_yamltests/matter_yamltests/runner.py

```diff
@@ -180,9 +172,6 @@ async def _run(self, parser: TestParser, config: TestRunnerConfig):
 
             test_duration = 0
             for idx, request in enumerate(parser.tests):
-                if request.is_event and request.event_number == 'newEventsOnly':
-                    request.event_number = self.last_event_number + 1
```

Comment:
Why is support for requesting only new events being removed?

Reply:
Or did I misunderstand what "request.event_number" does? I had assumed this would actually affect what gets sent in the request, but maybe it's actually just changing what checks we do on the response?

If so, then both the before and after code are wrong.  We should be excluding old events by setting a min event number in the _request_.

Reply:
This was the change from the PR I was replacing: this code is being replaced with the yaml file explicitly saying "lastEvent + 1".

Reply:
I am updating based on #28704 - event number should be indeed the minimum event number to be received. Since the tests pass I assumed that this works and is intentional. The logic intends to be: keep track of largest event number ever received then use that event number to be able to ask for "give me latest event(s)" only

Reply:
> Or did I misunderstand what "request.event_number" does? I had assumed this would actually affect what gets sent in the request, but maybe it's actually just changing what checks we do on the response?

You are correct in thinking that it affects what gets *sent*. It does set the `event-min` value.

Reply:
OK.  So in YAML, when it says `eventNumber` what that means is "set the EventMin in the request to this value"?  Is this documented anywhere @vivien-apple?  This would all be much easier to review if we had clear documentation for the expected behavior....

Reply:
Yes that is what it meant. 
For `chip-tool` case the workflow is:
 * `eventNumber` in yaml is translated to the `event_number` property at https://github.com/project-chip/connectedhomeip/blob/88f5ae4a2be59ab2a87ca086fb2f6ddfd7c20589/scripts/py_matter_yamltests/matter_yamltests/parser.py#L212
 * `event_number` property is translated by the `chip-tool` argument as `event-min`at https://github.com/project-chip/connectedhomeip/blob/1084dd53d50b54e52cc0d91a0f39f84dc68b7039/examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/encoder.py#L293
 * `event-min` is used by different command in `chip-tool` but afaik always translate into the `mEventNumber` property of the command to be set. One example is at see https://github.com/project-chip/connectedhomeip/blob/1084dd53d50b54e52cc0d91a0f39f84dc68b7039/examples/chip-tool/commands/clusters/ReportCommand.h#L319 
 * `mEventNumber` set the `mEventNumber` property of the `ReadPrepareParams` struct  at https://github.com/project-chip/connectedhomeip/blob/88f5ae4a2be59ab2a87ca086fb2f6ddfd7c20589/src/app/tests/suites/commands/interaction_model/InteractionModel.cpp#L465
 

This parameter does what it described by the spec, but I agree that it won't hurt to add some documentation to it. At least in `chip-tool`....
So no, this is not documented explicitly as far as I know

Reply:
We should name `eventNumber` to `eventMin` or similar.